### PR TITLE
Add columns `fmt_fn`, `warning`, and `errors` to `ard_attributes` output

### LINKED ISSUE #327: Add warning and error columns to `ard_attributes()` output
``` r
cards::ard_attributes(cards::ADSL[c("AGE", "AGEGR1")])
#> {cards} data frame: 4 x 5
#>   variable   context stat_name stat_label      stat
#> 1      AGE attribut…     label  Variable…       Age
#> 2      AGE attribut…     class  Variable…   numeric
#> 3   AGEGR1 attribut…     label  Variable… Pooled A…
#> 4   AGEGR1 attribut…     class  Variable… character
```

<sup>Created on 2024-09-23 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by jtalboys**
**What changes are proposed in this pull request?**
* Add columns `fmt_fn`, `warning`, and `errors` to `ard_attributes()` output (#327)

Provide more detail here as needed.

closes #327

Please let me know if there is anything else to change here. I'm not 100% sure I'm setting the `fmt_fn` column right? Also had a look at the docs for `ard_attributes()` to see if any changes were required but they only talk about the added attributes, don't mention the columns of the output.

--------------------------------------------------------------------------------

Pre-review Checklist (if item does not apply, mark is as complete)
- [x] **All** GitHub Action workflows pass with a :white_check_mark:
- [x] PR branch has pulled the most recent updates from master branch: `usethis::pr_merge_main()`
- [x] If a bug was fixed, a unit test was added.
- [x] Code coverage is suitable for any new functions/features (generally, 100% coverage for new code): `devtools::test_coverage()`
- [x] Request a reviewer

Reviewer Checklist (if item does not apply, mark is as complete)

- [x] If a bug was fixed, a unit test was added.
- [x] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [x] Code coverage is suitable for any new functions/features: `devtools::test_coverage()`

When the branch is ready to be merged:
- [x] Update `NEWS.md` with the changes from this pull request under the heading "`# cards (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [x] **All** GitHub Action workflows pass with a :white_check_mark:
- [x] Approve Pull Request
- [x] Merge the PR. Please use "Squash and merge" or "Rebase and merge".

### CODE REVIEW

File: R/ard_attributes.R [hunk 1]
```diff
@@ -82,6 +82,11 @@ ard_attributes.data.frame <- function(data,
       ),
       context = "attributes"
     ) |>
+    dplyr::mutate(
+      fmt_fn = list(as.character),
+      warning = list(NULL),
+      error = list(NULL)
+    ) |>
```
Comment (ddsjoberg) [inline R/ard_attributes.R:89]: This is great, thank you for adding! Can you move these to the `mutate()` that appears directly above?

Also for `fmt_fn`, your specification is perfect. But can we change it to the below? I was thinking about it more, and people may assign anything as an attribute so we can't assume it's a string. But we can assume the `label` is a string.

```r
fmt_fn = ifelse(.data$stat_name %in% "label", list(as.character), list(NULL))
```

**Review by ddsjoberg — CHANGES_REQUESTED** [review]
Thank you so much @jtalboys !! These look great! I just have one comment and the request below.

While reviewing I noticed that I had never added a check that the user-passed `label` argument is a named list.  Can you also add a check like this?

```r
  if (!is_empty(label)) {
    if (!is.list(label) || !is_named(label) || some(label, \(x) !is_string(x))) {
      cli::cli_abort(
        "The {.arg label} argument must be a named list with each element a string.",
        call = get_cli_abort_call()
      )
    }
  ```

File: R/ard_attributes.R (hunk 1 above)
Comment (jtalboys) [reply R/ard_attributes.R]: Moved the column creation to the `mutate()` above (not sure why I didn't just put them there in the first place) and used the above definition for `fmt_fn` - thank you!

**Review by ddsjoberg — APPROVED** [review]
This is wonderful, thank you @jtalboys !

**Pull request merged**